src/main/java/com/desiato/puresynth/services/UserService.java

```diff
@@ -42,8 +50,17 @@ public void deleteUser(Long id) {
     }
 
     public User createUser(String email, String password) {
-        User newUser = new User(email, password);
-        userRepository.save(newUser);
-        return newUser;
+        User newUser = new User(email, projectConfig.passwordEncoder().encode(password));
+        return userRepository.save(newUser);
+    }
+
+    public User updateUser(Long id, UserRequestDTO userRequestDTO) {
+        User existingUser = userRepository.findById(id)
```

Comment:
You should use a method `getUserByIdOrThrow` you created for that. Right now you have code duplication.